dotsql_test.go

```diff
@@ -322,10 +356,137 @@ func TestQueryContext(t *testing.T) {
 	}
 }
 
+func TestQueryOptions(t *testing.T) {
+	dot := DotSql{
+		queries: map[string]*template.Template{
+			"select": createTemplate(t, "SELECT * from users WHERE name = ?{{if  .exclude_deleted}} AND deleted IS NULL{{end}}"),
+		},
+	}
+	dotExcludeDeleted := dot.WithData(map[string]any{"exclude_deleted": true})
+
+	queryerStub := func(err error) *QueryerMock {
+		return &QueryerMock{
+			QueryFunc: func(_ string, _ ...any) (*sql.Rows, error) {
+				if err != nil {
+					return nil, err
+				}
+				return &sql.Rows{}, nil
+			},
+		}
+	}
+
+	testArg := "test"
+
+	// query not found
```

Comment:
each of these cases:
- query not found
- error returned by db
- error returned by db (with options)
- successful query (with options)

could be wrapped into `t.Run()` block, so that they are easy to understand and also if one case fails (via t.Fatal) the rest of the cases can run.